os/kernel/debug/memdbg.c

```diff
@@ -0,0 +1,102 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/***************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <debug.h>
+
+#include <tinyara/arch.h>
+#include <tinyara/mm/mm.h>
+
+/***************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Global Variables
+ ****************************************************************************/
+
+extern const uint32_t g_idle_topstack;
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: display_memory_information
+ *
+ * Description:
+ *   This function prints the memory information for debug such as
+ *   the RAM configuration, Idle stack configuration, Heap configuration.
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+
+void display_memory_information(void)
+{
+	/* Print RAM configuration */
+
+	mllvdbg("-------------------RAM Configuration--------------------\n");
+#ifdef CONFIG_SRAM_START_ADDR
+	mllvdbg("Internal RAM   : start = 0x%x size = %u \n", CONFIG_SRAM_START_ADDR, (CONFIG_KSRAM_SIZE + CONFIG_USRAM_SIZE));
+#if (defined(CONFIG_KSRAM_SIZE) && defined(CONFIG_RAM_KREGIONx_START))
```

Comment:
only START define check is enough?

Reply:
I think so , am i missing any other config check ?